libs/src/sdk/api/chats/ChatsApi.ts

```diff
@@ -131,6 +145,55 @@ export class ChatsApi extends BaseAPI {
     }
   }
 
+  public async listen() {
+    const timestamp = new Date().getTime()
+    const originalUrl = `/comms/chats/ws?timestamp=${timestamp}`
+    const signatureHeader = await this.getSignatureHeader(originalUrl)
+    const url = `ws://localhost:8925${originalUrl}&signature=${encodeURIComponent(
```

Comment:
how am i to get the selected DN here 🤔 ? Also how would I do it in a way that would be subject to the same health checks etc as the others? Do I just run it through the middleware manually here?

any thoughts @piazzatron ? @sliptype ?

Reply:
Another question is how do we want to handle reselections - does it disconnect your existing websocket?